sfputil/main.py

```diff
@@ -729,6 +738,134 @@ def eeprom_hexdump(port, page):
 
     click.echo(output)
 
+
+def dump_eeprom_for_all_ports(page):
+    lines = []
+
+    if page is not None:
+        try:
+            page = int(page)
+            # Only do simple validation to a given page, user is responsible to make sure the page existence
+            if page < 0 or page > 255:
+                click.echo(f"Invalid page {page}")
+                sys.exit(ERROR_INVALID_PAGE)
+        except ValueError:
+            click.echo(f"Page {page} is not a valid number")
+            sys.exit(ERROR_INVALID_PAGE)
+
+    for index, sfp in enumerate(platform_chassis.get_all_sfps()):
+        try:
+            presence = sfp.get_presence()
+            if not presence:
+                lines.append(f'\nModule {index + 1} not present')
+            else:
+                lines.append(f'\nEEPROM hexdump for module {index + 1}')
+                eeprom_data = dump_sfp_eeprom(sfp, page)
+                if eeprom_data is None:
+                    lines.append(f'{EEPROM_DUMP_INDENT}N/A\n')
+                else:
+                    lines.append(eeprom_data)
+        except NotImplementedError:
+            lines.append(f'\nModule {index + 1} not supported')
+        except Exception as e:
+            lines.append(f'\nModule {index + 1} get EEPROM failed: {e}')
+
+    click.echo('\n'.join(lines))
+
+
+def dump_sfp_eeprom(sfp, page):
+    api = sfp.get_xcvr_api()
+    if api is None:
+        return f'{EEPROM_DUMP_INDENT}N/A\n'
+
+    from sonic_platform_base.sonic_xcvr.api.public import sff8636, sff8436, cmis, sff8472
+    from sonic_platform_base.sonic_xcvr.fields import consts
+    if isinstance(api, cmis.CmisApi):
```

Comment:
@Junchao-Mellanox don't need to import this mid-way through the file, instead we can read the ID directly to know the API as done in existing code to choose between eeprom_hexdump_sff8472() and eeprom_hexdump_sff8636()

Reply:
Hi @prgeor , I think it is better way to do it like this. For example, say we have a new CMIS cable type whose ID is 0x34(just example). `if isinstance(api, cmis.CmisApi):` does not require any change; but reading ID solution will require a PR to change the sonic utilities to take the new ID 0x34 into account.

Reply:
I will double confirm that isinstance works for ccmis and other